modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -985,6 +985,95 @@ public void run() {
         StatusThreadPoolExecutor.getInstance().execute(runnableTask);
     }
 
+    @ReactMethod
+    public void multiformatSerializePublicKey(final String multiCodecKey, final String base58btc, final Callback callback) throws JSONException {
+        if (!checkAvailability()) {
+            callback.invoke(false);
+            return;
+        }
+
+        Runnable runnableTask = new Runnable() {
+            @Override
+            public void run() {
+                String res = Statusgo.multiformatSerializePublicKey(multiCodecKey,base58btc);
+                callback.invoke(res);
+            }
+        };
+
+        StatusThreadPoolExecutor.getInstance().execute(runnableTask);
+    }
+
+    @ReactMethod
+    public void multiformatDeserializePublicKey(final String multiCodecKey, final String base58btc, final Callback callback) throws JSONException {
```

Comment:
Not a suggestion for this PR, just an idea.

There are a bunch of methods here which require the same plumbing to execute the body of `run` in the thread pool. We could easily create a method that receives a lambda to remove all the duplication.

Reply:
yeah this is a good idea, I'll do this in a separate PR.
Link to issue : https://github.com/status-im/status-mobile/issues/14687